lighthouse-core/gather/driver/prepare.js

```diff
@@ -200,9 +200,13 @@ async function prepareTargetForIndividualNavigation(session, settings, navigatio
   /** @type {Array<LH.IcuMessage>} */
   const warnings = [];
 
-  const shouldResetStorage = !settings.disableStorageReset && !navigation.disableStorageReset;
+  const {requestor} = navigation;
+  const shouldResetStorage =
+    !settings.disableStorageReset &&
+    !navigation.disableStorageReset &&
+    typeof requestor === 'string';
```

Comment:
I feel like this divergent behavior on the type of `requestor` should be documented somewhere. Here? And should it be in user facing docs? Also on the `disableStorageReset` jsdoc?

Reply:
I added a note to the user flow docs. I'm not sure about calling it out on `disableStorageReset` jsdoc.